### ISSUE TITLE
REGEXP operator

### ISSUE THREAD

**Issue by ysard** [labsquare/cutevariant#169]
From #160 , it was badly named

New error:

```
INFO: [command.py:88:select_cmd()] command:select_cmd:: SELECT DISTINCT `variants`.`id`,`variants`.`favorite`,`variants`.`chr`,`variants`.`pos`,`variants`.`ref`,`variants`.`alt` FROM variants WHERE `variants`.`chr` REGEXP '1' LIMIT 50 OFFSET 0
ERROR: [mainwindow.py:213:refresh_plugins()] user-defined function raised exception
Traceback (most recent call last):
  File "/media/DATA/Projets/cutevariant/cutevariant/cutevariant/gui/mainwindow.py", line 211, in refresh_plugins
    plugin_obj.on_refresh()
  File "/media/DATA/Projets/cutevariant/cutevariant/cutevariant/gui/plugins/variant_view/widgets.py", line 846, in on_refresh
    self.load()
  File "/media/DATA/Projets/cutevariant/cutevariant/cutevariant/gui/plugins/variant_view/widgets.py", line 895, in load
    self.main_right_pane.load()
  File "/media/DATA/Projets/cutevariant/cutevariant/cutevariant/gui/plugins/variant_view/widgets.py", line 444, in load
    self.model.load()
  File "/media/DATA/Projets/cutevariant/cutevariant/cutevariant/gui/plugins/variant_view/widgets.py", line 221, in load
    having=self.having,
  File "/media/DATA/Projets/cutevariant/cutevariant/cutevariant/core/command.py", line 89, in select_cmd
    for i in conn.execute(query):
sqlite3.OperationalError: user-defined function raised exception
```

See the "SQL" function here: `REGEXP`

As it is explained here:
https://stackoverflow.com/questions/5071601/how-do-i-use-regex-in-a-sqlite-query

System package sqlite3-pcre must be installed on the system, which is not/can be not the case by default:
`sudo apt-get install sqlite3-pcre`

Also, old systems haven't this package in their repo:
https://packages.debian.org/fr/stretch/sqlite3-pcre

We should remove this operator from the project for better compatibility, support and easier installation process.
LIKE operator is more than enough for most cases, more users (biologists/PUPH/doctors) of this program are very far from mastering REGEX and having the full utility of such an advanced function.

TL;DR:
The system lib is not installed by default or not available.
It will crash on the quasi totality of the systems.

**Comment by dridk** [comment]
I am not sure it has anything to do with sqlite-pcre. 
It is a python function post processing. 
I am investigating your error

**Comment by dridk** [comment]
It works on windows ! Without extension requirements : 

![image](https://user-images.githubusercontent.com/1911063/95465543-91ff7f00-097b-11eb-828e-ba99fecd6e26.png)

**Comment by ysard** [comment]
Yes you only tested on str type fields and I only tested on numeric type fields.
Together we cover all possibilities ...

There are 2 errors here:
- Me who thought that it was necessary to use an external module
- The implementation which wrongly expects a str type

The external module I mentioned probably allows a big performance gain in comparison with the lambda that we implement in Python.

Moreover, the pre-compilation of the expression is unnecessary here because the lambda function is called for each entry of the tested table (potentially millions of times therefore).
I therefore remove the precompilation of the expression; this is an error that can be found in all tutorials and code fragments using the `create_function function` of the Python SQLite3 RDBMS.
In addition, each time the item must be cast in str.

This operator remains extremely expensive:
- no pre-compilation of regex
- systematic cast into str
- very frequent functional call

It is more expensive than LIKE which is very permissive here because it also works on numeric fields.
